_episodes/05-managing-cuda-dependencies.md

```diff
@@ -118,163 +112,84 @@ cudnn                          7.6.5       cuda9.2_0  pkgs/main
-library, used for applications such as computer vision and natural language processing. It is 
-primarily developed by Facebook’s AI Research lab. Conda is actually the 
-[recommended way](https://pytorch.org/get-started/locally/) to install PyTorch. The official 
-PyTorch binary ships with NCCL and cuDNN so it is not necessary to include these libraries in 
-your `environment.yml` file (unless some other package also needs these libraries!).
+We can check what packages have been installed using:
 
-~~~
-name: null
-
-channels:
-  - pytorch
-  - conda-forge
-  - defaults
-
-dependencies:
-  - cudatoolkit=10.1
-  - pip=20.0
-  - python=3.7
-  - pytorch=1.5
-  - torchvision=0.6
-~~~
-{: .language-yaml}
+```
+conda env export
+```
 
-> ## Check your channel priorities!
-> 
-> Also take note of the channel priorities: the official `pytorch` channel must be given priority 
-> over `conda-forge` in order to insure that the official PyTorch binaries (the ones that include 
-> NCCL and cuDNN) will be installed (otherwise you will get some unofficial version of PyTorch 
-> available on `conda-forge`).
-{: .callout}
+We might want to check this has worked:
+
+```
+python -c "import torch; print(torch.cuda.is_available())"
+```
 
 ## TensorFlow
 
-[TensorFlow](https://www.tensorflow.org/) is a free, open-source software library for dataflow and 
+[TensorFlow](https://www.TensorFlow.org/) is a free, open-source software library for dataflow and 
 differentiable programming across a range of tasks. It is a symbolic math library, and is also used 
-for machine learning applications such as neural networks. There are lots of versions and builds of 
-TensorFlow available via Conda (the output of `conda search tensorflow` is too long to share here!). 
+for machine learning applications such as neural networks.
+
+TensorFlow [recommend using `pip` to install](https://www.TensorFlow.org/install/pip), even if you're using Conda.
+
+As with pytorch, we'll make and activate a new environment:
 
-How do you decide which version is the “correct” version? How to make sure that you get a build 
-that includes GPU support? At this point you have seen all the Conda "tricks" required  to solve 
-this one yourself!
+```
+conda create --name tf-with-gpu
+conda activate tf-with-gpu
+```
 
-> ## Create an `environment.yml` file for TensorFlow
+Then follow the instructions to install TensorFlow e.g. on Windows 10:
+
+```
+conda install -c conda-forge cudatoolkit=11.2 cudnn=8.1.0
+# Anything above 2.10 is not supported on the GPU on Windows Native
+python3 -m pip install "tensorflow<2.11"
+```
+
+Note the lack of support for recent versions on Windows. The [Windows Subsystem for Linux](https://learn.microsoft.com/en-gb/windows/wsl/install)
+is a good choice for installing later versions.
+
+We might want to check this has worked:
+
+```
+python -c "import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))"
```

Comment:
As above, do you think we need to explain this in greater detail?

Reply:
Yup.